package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -1231,22 +1266,78 @@ def print(self, verbose, *args):
         elif verbose:
             print(args)
 
-    def run(self, tau_max=20, start=0, stop=None, step=1, verbose=False):
+    def _correct_intermittency(self, intermittency, selected_ids):
         """
-        Computes and returns the survival probability timeseries
+        Pre-process Consecutive Intermittency
+        :param intermittency: the max gap allowed and to be corrected
+        :param selected_ids: modifies the selecteded IDs in place by adding atoms which left for <= :param intermittency
+        """
+        self.print('Correcting the selected IDs for intermittancy (gaps). ')
```

Comment:
Was `self.print()` already in the class?

(We should be consistently using logging... but I think that's another PR for this module.)

Reply:
Yes, we've created self.print() in the previous PR. I'll look into logging for the next PR. Thanks